src/core/namespace.c

```diff
@@ -1048,11 +1091,9 @@ int setup_namespace(
                 if (r < 0)
                         goto finish;
 
-                if (namespace_info_mount_apivfs(root_directory, ns_info)) {
-                        r = append_static_mounts(&m, apivfs_table, ELEMENTSOF(apivfs_table), ns_info->ignore_protect_paths);
-                        if (r < 0)
-                                goto finish;
-                }
+                r = append_mount_apivfs(&m, root_directory, ns_info, ns_info->ignore_protect_paths);
+                if (r < 0)
+                        goto finish;
```

Comment:
> so, i am not sure I like this specific change, simply as a matter of being unnecessarily fine-grained. I am not sure it's worth optimizing in this area, I much rather would say that /proc, /proc and /dev being around is the absolute baseline of APIs to expect if you are a Linux C program. I mean, MountAPIVFS= is an option instead of the default primarily to keep compat with any status quo ante, not because it was really a good idea to make /proc and friends optional...

I do agree, for MountAPIVFS= of course, but for ProtectControlCgroups= and ProtectKernelTunables= I think they should not automatically bind /dev with the same write permissions inside unless MountAPIVFS= was set or PrivateDevices= where in this last case it is a private instance. I mean if the user sets ProtectKernelTunables= we make some /proc /sys readonly which is good but at same time we are bind mounting /dev inside which can also be used to directly write to memory... Thoughts ?